File: ios/Plugin/ZendeskSupportPlugin.swift
import Foundation
import Capacitor
import UIKit
/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ZendeskSupportPlugin)
public class ZendeskSupportPlugin: CAPPlugin, UINavigationControllerDelegate {
    private let implementation = ZendeskSupport()

    @objc func initialize(_ call: CAPPluginCall) {
        do {
            let appId = call.getString("appId") ?? ""
            let clientId = call.getString("clientId") ?? ""
            let zendeskUrl = call.getString("zendeskUrl") ?? ""
            let chatAccountKey = call.getString("chatAccountKey") ?? ""
            let messagingId = call.getString("iosChatId") ?? ""
            let debugLog = call.getBool("debugLog") ?? false
            try implementation.initialize( appId, clientId, zendeskUrl, messagingId)
            call.resolve()
        } catch {
            call.reject(error.localizedDescription, nil, error)
        }
    }

    @objc func setAnonymousIdentity(_ call: CAPPluginCall) {
        let name = call.getString("name") ?? ""
        let email = call.getString("email") ?? ""
        implementation.setAnonymousIdentity(name, email)
        call.resolve()
    }

    @objc func setIdentity(_ call: CAPPluginCall) {
        let token = call.getString("token") ?? ""
        implementation.setIdentity(token)
        call.resolve()
    }

    @objc func openChat(_ call: CAPPluginCall) {
        implementation.openChat(bridge?.viewController)
    }
}

File: android/src/main/java/com/websquare/plugins/zendesksupport/ZendeskSupportPlugin.kt
package com.websquare.plugins.zendesksupport

import com.getcapacitor.Plugin
import com.getcapacitor.annotation.CapacitorPlugin
import com.getcapacitor.PluginMethod
import com.getcapacitor.PluginCall
import java.lang.Exception

@CapacitorPlugin(name = "ZendeskSupport")
class ZendeskSupportPlugin : Plugin() {
    private val implementation = ZendeskSupport()
    @PluginMethod
    fun initialize(call: PluginCall) {
        val appId = call.getString("appId", "")
        val clientId = call.getString("clientId", "")
        val zendeskUrl = call.getString("zendeskUrl", "")
        val androidChatId = call.getString("androidChatId", "")
        val debugLog = call.getBoolean("debugLog", false)
        try {
            implementation.initialize(activity.applicationContext, zendeskUrl, appId, clientId, androidChatId, debugLog ?: false)
            call.resolve()
        } catch (e: Exception) {
            call.reject(e.message, e)
        }
    }

    @PluginMethod
    fun setAnonymousIdentity(call: PluginCall) {
        val name = call.getString("name", "")
        val email = call.getString("email", "")
        implementation.setAnonymousIdentity(name, email)
        call.resolve()
    }

    @PluginMethod
    fun setIdentity(call: PluginCall) {
        val token = call.getString("token", "")
        implementation.setIdentity(token)
        call.resolve()
    }

    @PluginMethod
    fun openChat(call: PluginCall) {
        implementation.openChat(activity)
        call.resolve()
    }
}
